l1-contracts/contracts/state-transition/StateTransitionManager.sol

```diff
@@ -96,33 +91,15 @@ contract StateTransitionManager is IStateTransitionManager, ReentrancyGuard, Own
         return SemVer.unpackSemVer(SafeCast.toUint96(protocolVersion));
     }
 
-    /// @notice Returns all the registered hyperchain addresses
-    function getAllHyperchains() public view override returns (address[] memory chainAddresses) {
-        uint256[] memory keys = hyperchainMap.keys();
-        chainAddresses = new address[](keys.length);
-        uint256 keysLength = keys.length;
-        for (uint256 i = 0; i < keysLength; ++i) {
-            chainAddresses[i] = hyperchainMap.get(keys[i]);
-        }
-    }
-
-    /// @notice Returns all the registered hyperchain chainIDs
-    function getAllHyperchainChainIDs() public view override returns (uint256[] memory) {
-        return hyperchainMap.keys();
-    }
-
-    /// @notice Returns the address of the hyperchain with the corresponding chainID
-    /// @param _chainId the chainId of the chain
-    /// @return chainAddress the address of the hyperchain
-    function getHyperchain(uint256 _chainId) public view override returns (address chainAddress) {
-        // slither-disable-next-line unused-return
-        (, chainAddress) = hyperchainMap.tryGet(_chainId);
+    /// @notice return the state transition chain contract for a chainId
+    function getHyperchain(uint256 _chainId) public view returns (address) {
+        return IBridgehub(BRIDGE_HUB).getHyperchain(_chainId);
     }
 
     /// @notice Returns the address of the hyperchain admin with the corresponding chainID
     /// @param _chainId the chainId of the chain
     function getChainAdmin(uint256 _chainId) external view override returns (address) {
```

Comment:
this method does not depend on STM at all, so it should be logically located here

Reply:
yeah, I meant that it shouldn't. Could you please add a comment on why it is there then